### PR TITLE
URL Params to hide tree and inspector

### PR DESCRIPTION
**Pull request by kobe1104**
Fixes https://github.com/nasa/openmct/issues/3758

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Testing instructions:
1. At openmct, add '&hideTree=true' and/or '&hideInspector=true' to the end of the url.
2. The corresponding pane collapses.
2. Refresh the page. The panes from step 2 stay collapsed and the params from step 1 persist.
3. Clicking the collapse button on the panes will add the hide params, and expanding the panes will remove hide params.

### CODE REVIEW

**Comment by kobe1104** [comment]
@nikhilmandlik The ApplicationRouterSpec tests are failing on this one as well. There's no tests added to this PR. It might have to do with the router calls I added.

**Comment by kobe1104** [comment]
This will be ready to review once the test issue caused by router is fixed.

**Comment by kobe1104** [comment]
Circle CI failing issue has been fixed now!

**Comment by davetsay** [comment]
@kobe1104, This should have unit tests

**Comment by kobe1104** [comment]
@davetsay I will add it!

File: src/ui/layout/pane.vue [hunk 1]
```diff
@@ -70,20 +71,55 @@ export default {
...
             } else {
-                // Pane is collapsed and is being expanded
-                this.$el.style[this.styleProp] = this.currentSize;
-                delete this.currentSize;
-                delete this.dragCollapse;
+                this.handleExpand();
+            }
+        },
+        handleHideUrl: function () {
+            const HIDE_TREE_PARAM = 'hideTree';
```
Comment (shefalijoshi) [inline src/ui/layout/pane.vue:90]: Let's move these consts outside of the method at the top of the script.

File: src/ui/layout/pane.vue [hunk 2]
```diff
@@ -70,20 +71,55 @@ export default {
...
+            for (let key of this.$el.classList) {
+                if (key === PaneClass && this.collapsable) {
+                    this.collapsed = true;
+                    this.handleCollapse();
+                    this.openmct.router.deleteSearchParam(param);
+                    break;
+                }
+            }
+        },
+        handleCollapse: function () {
```
Comment (shefalijoshi) [inline src/ui/layout/pane.vue:114]: Can we use the existing `toggleCollapse` method here instead?

**Review by shefalijoshi — CHANGES_REQUESTED** [review]

**Comment by kobe1104** [comment]
Per discussion with @davetsay and @charlesh88, we all agreed that the URL params that hide the panes should be persisted. In case users bookmarks the URL, they should see exactly the same view. 
I will make the URL params toggle when user make the pane collapse or expand it.

File: src/ui/layout/pane.vue (hunk 1 above)
Comment (kobe1104) [reply src/ui/layout/pane.vue]: Hi @shefalijoshi I've moved them.

File: src/ui/layout/pane.vue (hunk 2 above)
Comment (kobe1104) [reply src/ui/layout/pane.vue:112]: Hi @shefalijoshi  the toggleCollapse handles both collapse and expand. It calls either handleCollapse or handleExpand.

**Review by shefalijoshi — APPROVED** [review]
LGTM

**Comment by shefalijoshi** [comment]
@kobe1104 Please move the Testing Instructions to the Issue.

**Pull request merged**